sdks/typescript/src/apache_beam/utils/service.ts

```diff
@@ -326,14 +330,37 @@ export class JavaJarService extends SubprocessService {
     }
   }
 
-  static mavenJarUrl(
+  static async mavenJarUrl(
     artifactId: string,
     version: string,
     classifier: string | undefined = undefined,
     appendix: string | undefined = undefined,
     repo: string = JavaJarService.APACHE_REPOSITORY,
     groupId: string = JavaJarService.BEAM_GROUP_ID
-  ): string {
+  ): Promise<string> {
+    if (version == "latest") {
+      const medatadataUrl = [
+        repo,
+        groupId.replaceAll(".", "/"),
+        artifactId,
+        "maven-metadata.xml",
+      ].join("/");
+      const metadata = await new Promise<string>((resolve, reject) => {
+        let data = "";
+        https.get(medatadataUrl, (res) => {
+          res.on("data", (chunk) => {
+            data += chunk;
+          });
+          res.on("end", () => {
+            resolve(data);
+          });
+          res.on("error", (e) => {
+            reject(e);
+          });
+        });
+      });
```

Comment:
Optional: It might be worth considering a http client library that can do things like handle retries/redirects and will patch responses together for us - I'm partial to https://github.com/microsoft/typed-rest-client (the Http client, not the Rest client in this case), but there are plenty of options out there.

Reply:
I'm going to avoid taking on another dependency right now, but that's good to keep in mind if we end up needing to make these kinds of calls more often.

Reply:
Sounds good